components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/GeneListController.java

```diff
@@ -183,4 +189,78 @@ public void writeJSON(Patient patient, JSONObject json, Collection<String> selec
             }
         }
     }
+
+    @Override
+    public PatientData<Map<String, String>> readJSON(JSONObject json)
+    {
+        Object testGeneJson = json.get(this.getJsonPropertyName());
+        if (testGeneJson == null) {
+            return null;
+        }
+
+        try {
+            JSONArray genesJson = json.getJSONArray(this.getJsonPropertyName());
+            List<Map<String, String>> allGenes = new LinkedList<Map<String, String>>();
+            for (Object geneJsonUncast : genesJson) {
+                JSONObject geneJson = JSONObject.fromObject(geneJsonUncast);
+                Map<String, String> singleGene = new LinkedHashMap<String, String>();
+                for (String property : getProperties()) {
+                    // todo. it makes no sense to have no gene name, but to have comments, which can occur
```

Comment:
I don't get the comment :) If it makes no sense, then when does it occur, and should we support it, and if yes why?